crates/iroha/tests/fees.rs

```diff
@@ -0,0 +1,89 @@
+use eyre::Result;
+use fees_executor_data_model::parameters::*;
+use iroha::data_model::prelude::*;
+use iroha_executor_data_model::parameter::Parameter as _;
+use iroha_test_network::*;
+mod upgrade;
+use upgrade::upgrade_executor;
+
+#[test]
+fn fees_options_in_updated_executor() -> Result<()> {
+    let (network, _rt) = NetworkBuilder::new().start_blocking()?;
+
+    let test_client = network.client();
+    // Also sets enough fuel based on the profile
+    upgrade_executor(&test_client, "fees_executor")?;
+
+    let parameters = test_client.query_single(FindParameters)?;
+    let fees_options: FeesOptions = parameters
+        .custom()
+        .get(&FeesOptions::id())
+        .unwrap()
+        .try_into()
+        .unwrap();
+
+    assert_eq!(fees_options, FeesOptions::default());
+
+    Ok(())
+}
+
+#[test]
+fn fees_options_cannot_change() -> Result<()> {
+    let (network, _rt) = NetworkBuilder::new().start_blocking()?;
+
+    let test_client = network.client();
+    upgrade_executor(&test_client, "fees_executor")?;
+
+    let result = test_client.submit_blocking(SetParameter::new(Parameter::Custom(
+        FeesOptions::default().into(),
+    )));
+    assert!(result.is_err());
```

Comment:
Is it possible to assert that the error contains the message string from the executor? E.g. "Fees cannot change".